TensorLib/Common.lean

```diff
@@ -338,49 +338,66 @@ def _root_.ByteArray.toBV64 (x : ByteArray) (startIndex : Nat) : Err BV64 :=
 The largest Nat such that it and every smaller Nat can be represented exactly in a 64-bit IEEE-754 float
 https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER
 
-The number is one less than 2^(mantissa size)
+The number 2^(mantissa size).
+
 https://en.wikipedia.org/wiki/Floating-point_arithmetic
 https://en.wikipedia.org/wiki/Double-precision_floating-point_format
 -/
-private def floatMantissaBits : Nat := 52
 private def float32MantissaBits : Nat := 23
+private def float64MantissaBits : Nat := 52
 -- Add 1 to the mantissa length because of the implicit leading 1
-def maxSafeNatForFloat : Nat := Nat.pow 2 (floatMantissaBits + 1) - 1
-def maxSafeNatForFloat32 : Nat := Nat.pow 2 (float32MantissaBits + 1) - 1
+def maxSafeNatForFloat32 : Nat := Nat.pow 2 (float32MantissaBits + 1)
```

Comment:
-1 was too pessimistic; we can obviously represent 2^24 and 2^53 exactly; 0 mantissa, int exponent.